### PR TITLE
Use minimal travis container

### PR DESCRIPTION
**Pull request by tv3141**
For testing with Travis we install Python using miniconda and only use this version of Python. Specifying `- python: 3.6` in `.travis.yml` sets the version of Python that comes installed in the Travis container.

Instead:
- use minimal travis container
- specify Python version as env var

### CODE REVIEW

**Comment by bjlittle** [comment]
@tv3141 Nice, I like it :+1: 

I'd hope that `minimal` builds spin up faster as well :wink:

File: .travis.yml [hunk 1]
```diff
@@ -5,18 +5,19 @@
...
-    - TEST_TARGET=default
-    - TEST_TARGET=example
-    - TEST_TARGET=doctest
+    - PYTHON_VERSION=2.7 TEST_TARGET=default TEST_MINIMAL=true
+    - PYTHON_VERSION=3.6 TEST_TARGET=default TEST_MINIMAL=true
+    - PYTHON_VERSION=2.7 TEST_TARGET=default
+    - PYTHON_VERSION=3.6 TEST_TARGET=default
+    - PYTHON_VERSION=2.7 TEST_TARGET=example
+    - PYTHON_VERSION=3.6 TEST_TARGET=example
+    - PYTHON_VERSION=3.6 TEST_TARGET=doctest
```
Comment (bjlittle) [inline .travis.yml]: Shame we have to crank the matrix permutations by hand...

**Comment by tv3141** [comment]
> @tv3141 Nice, I like it 
> 
> I'd hope that `minimal` containers spin up faster as well

@bjlittle I thought so as well, but run times vary wildly. The main benefit is that CI won't rely on something in the linux image that is not used. Adding Python3.7 to the linux image caused some problems https://github.com/travis-ci/travis-ci/issues/9069

**Comment by tv3141** [comment]
Close in preference of #3222

**Comment by bjlittle** [comment]
@tv3141 Awesome, thanks!

**Pull request closed without merging**